### DISCUSSION TITLE
Not sure whether I need `regenerator-runtime` as a dependency in my project

### DISCUSSION THREAD

**Discussion by ChristophP** [parcel-bundler/parcel#9182] [Q&A]
Before Parcel v2, I used Parcel v1 along with babel to transpile my JS code. It needed `core-js` for a bunch of polyfills and `regenerator-runtime` as to transpile async functions. I couldn't find a sufficient clarification on this in the docs.

Since Parcel v2, I got a bunch of messages that babel is not needed if I only need transpilation for certain things, which fit my case. I removed babel and relied on the transpilation offered by (I assume) swc alone.
This works well, However I recently noticed that I still have `regenerator-runtime` (and `core-js` for that matter) in my dependencies which is not imported anywhere in my source code (but previously dynamically injected by babel IIRC).

So my question due to lacking understanding of what parcel does under the hood: If I'm using async function, transpiling for the browser and use no babel rc with my parcel v2 setup, do I still need `regenerator-runtime` as a dependency?

Thanks in advance for the clarification.

**Comment by mischnic** [comment]
> do I still need regenerator-runtime as a dependency?

I think the answer is no, [swc uses its own helper for transpiling async/await](https://play.swc.rs/?version=1.3.74&code=H4sIAAAAAAAAA0ssrsxLVkgrzUsuyczPU6jQ0KzmUgCCxPLEzBKFSmuuWgDCndv8IgAAAA%3D%3D&config=H4sIAAAAAAAAA22PSw7CMAxE95yi8potLDgBGw5hBRelyk%2B2K7WqeneStCkgsYtn3mTs5dR1MIiBW7fkZx4SshAfc1ZkDopTVoCMRzFsk8K5uYMUq0cnVKV1c0CRX6Q1JZcdBxejUMN3zdtg%2B%2Fm70ESfmER%2BwYJieDn6W0eTEgd0d3KJuESVx4JUAHx8jjW5X6lzom23K3ygtsnRClYeLVn%2FW98mTB8tMQEAAA%3D%3D)

[Parcel 2 uses swc with `mode: "entry"` for core-js](https://github.com/parcel-bundler/parcel/blob/5ac45672bb603841bf294c66019930be4354625a/packages/transformers/js/core/src/lib.rs#L290-L294), so if you write `import "core-js"; ` in your code, [then that will be expanded. ](https://play.swc.rs/?version=1.3.74&code=H4sIAAAAAAAAA0ssrsxLVkgrzUsuyczPU6jQ0KzmUgCCxPLEzBKFSmuuWgDCndv8IgAAAA%3D%3D&config=H4sIAAAAAAAAA22PSw7CMAxE95yi8potLDgBGw5hBRelyk%2B2K7WqeneStCkgsYtn3mTs5dR1MIiBW7fkZx4SshAfc1ZkDopTVoCMRzFsk8K5uYMUq0cnVKV1c0CRX6Q1JZcdBxejUMN3zdtg%2B%2Fm70ESfmER%2BwYJieDn6W0eTEgd0d3KJuESVx4JUAHx8jjW5X6lzom23K3ygtsnRClYeLVn%2FW98mTB8tMQEAAA%3D%3D)

**Reply by ChristophP** [reply]
Awesome, thank you. I will remove the uneeded parts
